PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx

```diff
@@ -186,112 +134,111 @@ Bool_t AliAnalysisTaskJetCharge::FillHistograms()
       return 0;
   }
 
+  // Reset the Tree Parameters
+  Pt = 0;
+  Phi = 0;
+  Eta = 0;
+  JetCharge = 0;
+  LeadingTrackPt = 0;
 
+  // Initialise jet pointer
+  //cout << "Running Fill Histograms" << endl;
   AliEmcalJet *Jet1 = NULL; //Original Jet in the event                                                                                                         // Get jet container (0 = ?)
   AliJetContainer *JetCont= GetJetContainer(0); //Jet Container for event
-  Double_t JetPhi=0;
-  Double_t JetPt_ForThreshold=0;
+  Int_t nAcceptedJets = JetCont->GetNAcceptedJets();
+
+  //TClonesArray *trackArr = dynamic_cast<TClonesArray*>(InputEvent()->FindListObject("HybridTracks"));
+  Float_t JetPhi=0;
+  Float_t JetPt_ForThreshold=0;
+
   if(JetCont) {
     // Technical detail; fix possibly corrupted jet container ID
     JetCont->ResetCurrentID();
     // Jet is acceptable?
     while((Jet1=JetCont->GetNextAcceptJet())) {

```

Comment:
I guess this was already here, but be aware this is basically a deprecated way of accessing the jets. Preferred is:

```c++
for (auto Jet1 : JetCont->accepted()) {
  // ...
}
```

Reply:
You need to remove the `ResetCurrentID` part too. This can some issues in some causes, which is part of why it was deprecated